src/ralph/backends/data/base.py

```diff
@@ -144,25 +144,65 @@ def list(
         """
 
 
-class BaseDataBackend(ABC):
+def get_backend_generic_argument(backend_class: Type, position: int) -> Optional[Type]:
+    """Return the generic argument of `backend_class` at specified `position`."""
+    if not hasattr(backend_class, "__orig_bases__"):
+        return None
+
+    bases = backend_class.__orig_bases__[0]
+    if not hasattr(bases, "__args__") or len(bases.__args__) < abs(position) + 1:
+        return None
+
+    argument = bases.__args__[position]
+    if argument is Any:
+        return None
+
+    if isinstance(argument, TypeVar):
+        return argument.__bound__
+
+    return argument
+
+
+def set_backend_settings_class(backend_class: Type):
+    """Set `settings_class` attribute with `Config.env_prefix` for `backend_class`."""
```

Comment:
Currently, we don't alter the `backend_class`. However, we could automatize setting the `Config.env_prefix` here with something similar to:
```python
settings_class.Config.env_prefix = f"RALPH_BACKENDS__DATA__{backend_class.name.upper()}__"
```
or (to avoid altering the class):
```python
class Config(BaseSettingsConfig):
    """Pydantic Configuration."""

    env_prefix = f"RALPH_BACKENDS__DATA__{backend_class.name.upper()}__"
    env_file = ".env"
    env_file_encoding = core_settings.LOCALE_ENCODING

backend_class.settings_class = type(
    settings_class.__name__, (settings_class,), {"Config": Config}
)
```
However, then, we will have separate environment variables for configuring sync and async data backends. Ex:
```
RALPH_BACKENDS__DATA__ES__DEFAULT_INDEX=statements
RALPH_BACKENDS__DATA__ASYNC_ES__DEFAULT_INDEX=statements
```
WDYT?